lib/s3.py

```diff
@@ -95,6 +100,26 @@ def sign_curl(url: urllib.parse.ParseResult, method='GET', checksum=SHA256_NIL,
     return [f'-H{key}:{value}' for key, value in headers.items()] + [url.geturl()]
 
 
+def urlopen(url: urllib.parse.ParseResult, method='GET') -> IO:
+    """Same as sign_request() but calls urlopen() on the result"""
+    headers = sign_request(url, method=method)
+    request = urllib.request.Request(url.geturl(), headers=headers, method=method)
+    return urllib.request.urlopen(request)
+
+
+def list_bucket(url: urllib.parse.ParseResult) -> ET:
+    """Get the ListBucketResult as a xml.etree.ElementTree"""
+    with urlopen(url) as response:
+        return ET.fromstring(response.read())
+
+
+def parse_list(result: ET, *keys) -> Iterable[Iterable[str]]:
+    """For each item in the bucket, return the given keys"""
+    ns = {'s3': 'http://s3.amazonaws.com/doc/2006-03-01/'}
```

Comment:
That.. needs an explanation. Without authentication, this URL is "access denied", and if it's authenticated then the http:// isn't exactly inspiring trust. Can you please add a comment?

Reply:
That's just an XML namespace.  I'll add a comment.